### PR TITLE
kopf/structs/hierarchies.py: handle apiVersion attribute inconsistene across api resources

### PR DESCRIPTION
**Pull request by olivier-mauras**
# One-line summary
handle apiVersion attribute inconsistene across api resources
> Issue : #166 

## Description
The attribute apiVersion can be api_version depending of api resource and objects.

## Types of Changes
- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: kopf/structs/hierarchies.py [hunk 1]
```diff
@@ -8,8 +8,14 @@ def build_object_reference(body):
     """
     Construct an object reference for the events.
     """
+    # apiVersion attribute is not consistent across api resources
+    if 'apiVersion' in body:
```
Comment (psycho-ir) [inline kopf/structs/hierarchies.py]: Thank you so much for contributing to this project. 
Would be great to extract this code to a function to reduce the duplication.

Comment (olivier-mauras) [reply kopf/structs/hierarchies.py]: Done

Comment (psycho-ir) [reply kopf/structs/hierarchies.py]: Thanks @olivier-mauras, 
Would also be great if you could add one or more resource names as example for each attribute name.

Comment (olivier-mauras) [reply kopf/structs/hierarchies.py]: I added a  comment https://github.com/zalando-incubator/kopf/issues/166#issuecomment-519126078 on the bug I opened.
It's actually not the API resources that differ but the body returned by on.xxxx() handlers.

**Comment by nolar** [comment]
@olivier-mauras Thanks for your reported issue and contribution. Hopefully, there will be more in the future.

_For historic reference:_ The reason why this PR and this fix are not accepted is explained in https://github.com/zalando-incubator/kopf/issues/166#issuecomment-519171462. **TL;DR:** It is the Kubertenes Client's specific convention of snake-cased `api_version`; the K8s API provides and expects the camel-cased `apiVersion` (same is true for other multi-word fields).

**Pull request closed without merging**